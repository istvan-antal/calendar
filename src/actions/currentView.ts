import { createAction, ActionsUnion } from '.';

export const currentViewActions = {
    receiveCurrentTime: (currentTime: string) => createAction('currentView/receiveCurrentTime', currentTime),
};

export type CurrentViewAction = ActionsUnion<typeof currentViewActions>;