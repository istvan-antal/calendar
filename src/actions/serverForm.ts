import { createAction, ActionsUnion } from '.';

export const serverFormActions = {
    setServer: (server: string) => createAction('serverForm/setServer', server),
    setUsername: (username: string) => createAction('serverForm/setUsername', username),
    setPassword: (password: string) => createAction('serverForm/setPassword', password),
    open: () => createAction('serverForm/open'),
    save: () => createAction('serverForm/save'),
    cancel: () => createAction('serverForm/cancel'),
};

export type ServerFormActions = ActionsUnion<typeof serverFormActions>;