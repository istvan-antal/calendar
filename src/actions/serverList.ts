import { createAction, ActionsUnion } from '.';

interface Server {
    username: string;
    password: string;
    server: string;
}

export const serverListActions = {
    receiveServers: (servers: Server[]) => createAction('receiveServers', servers),
};

export type ServerListActions = ActionsUnion<typeof serverListActions>;