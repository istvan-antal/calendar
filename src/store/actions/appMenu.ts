import { createAction, ActionsUnion } from '.';

export const appMenuActions = {
    open: () => createAction('appMenu/open'),
    close: () => createAction('appMenu/close'),
    toggle: () => createAction('appMenu/toggle'),
};

export type AppMenuAction = ActionsUnion<typeof appMenuActions>;