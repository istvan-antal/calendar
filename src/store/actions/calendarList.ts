import { createAction, ActionsUnion } from '.';
import { Event } from '../../util/vcalendar';

export interface Calendar {
    url: string;
    displayName: string;
    events: Event[];
}

export const calendarListActions = {
    open: () => createAction('calendarList/open'),
    close: () => createAction('calendarList/close'),
    receive: (calendars: Calendar[]) => createAction('calendarList/receive', calendars),
    receiveCalendarColors: (colors: {
        [key: string]: string;
    }) => createAction('calendarList/receiveCalendarColors', colors),
    setCalendarColor: ({ calendar, color }: { calendar: Calendar; color: string }) => createAction(
        'calendarList/setCalendarColor',
        { calendar, color },
    ),
};

export type CalendarListAction = ActionsUnion<typeof calendarListActions>;