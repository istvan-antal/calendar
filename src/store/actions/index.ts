export interface Action<T extends string> {
    type: T;
}

export interface ActionWithData<T extends string, P> extends Action<T> {
    payload: P;
}

// tslint:disable:only-arrow-functions
export function createAction<T extends string>(type: T): Action<T>;
export function createAction<T extends string, P>(type: T, payload: P): ActionWithData<T, P>;
export function createAction<T extends string, P>(type: T, payload?: P) {
    return payload === undefined ? { type } : ({ type, payload });
}

// tslint:disable-next-line:no-any
type FunctionType = (...args: any[]) => any;
interface ActionCreatorMapObject { [actionCreator: string]: FunctionType; }
export type ActionsUnion<A extends ActionCreatorMapObject> = ReturnType<A[keyof A]>;