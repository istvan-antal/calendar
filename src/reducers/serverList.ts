import { ServerListActions } from '../actions/serverList';

export const serverList = (state = [], action: ServerListActions) => {
    switch (action.type) {
    default:
        return state;
    }
};
