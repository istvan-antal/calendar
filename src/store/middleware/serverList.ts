import { MiddlewareAPI, Dispatch, AnyAction } from 'redux';
import { State } from '../../reducers';
import { ServerListActions } from '../../actions/serverList';

export const serverListMiddleware = (store: MiddlewareAPI<Dispatch<ServerListActions | AnyAction>, State>) => (
    next: Dispatch<ServerListActions>,
) => (action: ServerListActions) => {
    const result = next(action);

    switch (action.type) {
        default:
            break;
    }

    return result;
};
