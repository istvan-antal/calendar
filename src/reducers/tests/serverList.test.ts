import { serverList } from '../serverList';
import { serverListActions } from '../../actions/serverList';

test('serverList', () => {
    // tslint:disable-next-line:no-any
    expect(serverList(undefined, { type: '' } as any)).toMatchSnapshot();
    expect(serverList(undefined, serverListActions.receiveServers([]))).toMatchSnapshot();
    expect(serverList([], serverListActions.receiveServers([]))).toMatchSnapshot();
});