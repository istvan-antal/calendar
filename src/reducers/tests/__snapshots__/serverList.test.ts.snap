// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serverList 1`] = `Array []`;

exports[`serverList 2`] = `Array []`;

exports[`serverList 3`] = `Array []`;
